catalog/format/iceberg/src/main/java/org/projectnessie/catalog/formats/iceberg/rest/IcebergMetadataUpdate.java

```diff
@@ -234,7 +235,7 @@ interface SetStatistics extends IcebergMetadataUpdate {
     default void applyToTable(IcebergTableMetadataUpdateState state) {
       long snapshotId = Objects.requireNonNull(state.snapshot().icebergSnapshotId());
       if (snapshotId == snapshotId()) {
-        state.builder().addStatisticsFile(icebergStatisticsFileToNessie(statistics()));
+        state.builder().statisticsFiles(singleton(icebergStatisticsFileToNessie(statistics())));
```

Comment:
Similar problem exist for partition statistics file below at line 274. 

Also, how can we catch them with tests? Is the unit tests missing?